qiskit/primitives/backend_estimator_v2.py

```diff
@@ -0,0 +1,262 @@
+    precludes doing any provider- or backend-specific optimizations.
+
+    Implementation of :class:`BaseEstimatorV2` using a backend.
+
+    This class provides a EstimatorV2 interface from any :class:`~.BackendV2` backend
+    and doesn't do any measurement mitigation, it just computes the expectation values.
+    At present, this implementation is only compatible with Pauli-based observables.
+
+    Each tuple of ``(circuit, observables, <optional> parameter values, <optional> precision)``,
+    called an estimator primitive unified bloc (PUB), produces its own array-based result. The
+    :meth:`~.EstimatorV2.run` method can be given a sequence of pubs to run in one call.
+    """
+
+    _VARIANCE_UPPER_BOUND: float = 4.0
+
+    def __init__(
+        self,
+        *,
+        backend: BackendV2,
+        default_precision: float = 0.0,
+        abelian_grouping: bool = True,
+    ):
+        """
+        Args:
+            backend: Required: the backend to run the primitive on
+            default_precision: Default precision
+            abelian_grouping: Whether the observable should be grouped into commuting
+        """
+        super().__init__()
+        self._backend = backend
+        basis = PassManagerConfig.from_backend(backend).basis_gates
+        self._passmanager = PassManager(
+            [Optimize1qGatesDecomposition(basis=basis, target=backend.target)]
+        )
+        self._default_precision = default_precision
+        self._abelian_grouping = abelian_grouping
+
+    @property
+    def default_precision(self) -> float:
+        """Return the default precision"""
+        return self._default_precision
+
+    @property
+    def ablian_grouping(self) -> bool:
+        """Return whether Abelian grouping is used or not."""
+        return self._abelian_grouping
+
+    @property
+    def backend(self) -> BackendV2:
+        """Returns the backend which this sampler object based on."""
+        return self._backend
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], *, precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        if precision is None:
+            precision = self._default_precision
+        coerced_pubs = [EstimatorPub.coerce(pub, precision) for pub in pubs]
+        self._validate_pubs(coerced_pubs)
+        job = PrimitiveJob(self._run, coerced_pubs)
+        job._submit()
+        return job
+
+    def _validate_pubs(self, pubs: list[EstimatorPub]):
+        for i, pub in enumerate(pubs):
+            if pub.precision == 0.0:
+                raise ValueError(
+                    f"The {i}-th pub has precision 0. But precision should be larger than 0."
+                )
+
+    def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
+        return PrimitiveResult([self._run_pub(pub) for pub in pubs])
+
+    def _run_pub(self, pub: EstimatorPub) -> PubResult:
+        shots = int(np.ceil(self._VARIANCE_UPPER_BOUND / pub.precision**2))
```

Comment:
Since I'm not sure the details of `precison`, this `shots` estimation is quite rough one. Could you tell me how can we determine the number of shots depending on `precision`, @ihincks and @chriseclectic?

Reply:
I will update it with the same default value with qiskit-ibm-runtime.
https://github.com/Qiskit/qiskit-ibm-runtime/blob/50797a909f376a5ab051b4cefd787ced7a2160e0/qiskit_ibm_runtime/options/estimator_options.py#L44

Reply:
Thank you for the details. I copied the API doc of qiskit-ibm-runtime and simplified the code by removing `_VARIANCE_UPPER_BOUND`.